lib/TsconfigPathsPlugin.js

```diff
@@ -0,0 +1,238 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Natsu @xiaoxiaojx
+*/
+
+"use strict";
+
+const path = require("path");
+const { aliasResolveHandler } = require("./AliasUtils");
+const { modulesResolveHandler } = require("./ModulesUtils");
+
+/** @typedef {import("./Resolver")} Resolver */
+/** @typedef {{alias: string | string[] | false, name: string, onlyModule?: boolean}} AliasOption */
+/** @typedef {import("./Resolver").ResolveStepHook} ResolveStepHook */
+
+/**
+ * @typedef {object} TsconfigPathsPluginOptions
+ * @property {string=} configFile - Path to tsconfig.json file
+ * @property {string=} baseUrl - Override baseUrl from tsconfig.json
+ * @property {string=} context - Context directory for resolving relative paths
+ */
+
+module.exports = class TsconfigPathsPlugin {
+	/**
+	 * @param {string} configFile tsconfig file path
+	 */
+	constructor(configFile) {
+		this.configFile = configFile;
```

Comment:
Ideally we should accept just `true` option too (i.e. `string` and `true`, maybe better name for `true` is `auto`, what do you think?), it means try to load tsconfig, if not found do nothing, for better development experience, setting paths to the config file can takes a time and sometimes it's not very convenient